dashboard/src/pages/[workspaceSlug]/[appSlug]/settings/general/index.tsx

```diff
@@ -69,6 +70,10 @@ export default function SettingsGeneralPage() {
   const { register, formState } = form;
 
   const handleProjectNameChange = async (data: ProjectNameValidationSchema) => {
+    await router.push({
```

Comment:
This is the bit that solve the `404` flashing and the problem with changing the application name redirecting to 404 (which I just found out) basically, if we are in the process of mutating we change the browser URL to contain a query like `?mutating=true` this is because as soonest as we change the slug, the apollo cache realizes that there isn't a workspace/application with that slug and redirects to `404` .  
  
Alternatives would be to rewrite to the apollo [https://www.apollographql.com/docs/react/caching/cache-interaction/#using-graphql-queries](https://www.apollographql.com/docs/react/caching/cache-interaction/#using-graphql-queries) (which would be kind-of a more correct implementation) but we have been staying away from this given that if we move away from Apollo at some point, we would only have queries/mutations (it would solve it as long as a re-fetch doesn't happen in the meantime, given that our main hooks like `currentWorkspace..` fetchs the entire user workspace/app information.)   
  
Alternatives to keeping state in the URL would be to just use context so we don't show anything in the URL even if for a few seconds. Which may not be that bad since we also have to solve a few things regarding the last workspace used (so when creating a new workspace the user can use the last one he used instead of the same every time.) Though we may have to be careful about possible side-effects. 

wdyt? @szilarddoro

Reply:
This is a good one @guicurcio. I like this approach. Maybe one thing I would change is `mutating` to `updating` or something like that.